### PR TITLE
Use triangles for vectors

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR changes the rendering of the vectors from openGL lines to be mesh objects made from triangles. It makes no changes to the API or functionality, but should prevent a bug around setting lines to arbitrary widths that we had encountered using vispy Lines and a bug around our osx build detailed in #213 

It should make it easy in a follow-up PR to add support for arrowheads if desired.

Finally adding an vectors layer to an empty viewer will cause an error as detailed in #214, but that is a broader issue that can be fixed later.

@bryantChhun do you want to give this a check to make sure all functionality is preserved and is behaving as expected. 

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [x] run `examples/add_vectors_coords.py` and `examples/add_vectors_image.py`

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by kne42** [comment]
Looks like OSX is still failing – does this mean that there's another bug somewhere?

**Comment by sofroniewn** [comment]
@kne42 it's unclear the osx 3.7 now passes which i didn't before, and the test failed at the `add_vectors_image` and not `add_image_coords` so something has improved, but unclear what is failing now. I am re-running the failed test

**Comment by sofroniewn** [comment]
OSX test passed now - not necessarily a great sign that it failed once, but I'll take the improvement

File: examples/add_vectors_coords.py
Comment (jni) [inline examples/add_vectors_coords.py]: @sofroniewn can you remove these print statements now?

File: examples/add_vectors_image.py
Comment (jni) [inline examples/add_vectors_image.py]: Generally I prefer `np.random.random((24, 100))` as it is consistent with other NumPy functions (zeros, ones, full, etc).

File: napari/layers/_vectors_layer/model.py
Comment (jni) [inline napari/layers/_vectors_layer/model.py:396]: offests -> offsets

@sofroniewn I thought in your shapes PR you added some machinery for drawing lines — wouldn't it be better to reuse that machinery here? Or is there a reason why this is different?

File: napari/layers/_vectors_layer/model.py
Comment (jni) [inline napari/layers/_vectors_layer/model.py]: This should go into some `vector_utils` module, or be made private, or both.

**Review by jni — APPROVED** [review]
@sofroniewn some minor comments, approved pending those. Would love to know the answer to my mesh reuse question before it gets merged though! =)

File: examples/add_vectors_coords.py
Comment (sofroniewn) [reply examples/add_vectors_coords.py]: done

File: examples/add_vectors_image.py
Comment (sofroniewn) [reply examples/add_vectors_image.py]: fixed

File: napari/layers/_vectors_layer/model.py
Comment (sofroniewn) [reply napari/layers/_vectors_layer/model.py:396]: so the only part that is in common right now is the  `segment_normal` function. I'm going to add that to our model general utils.

Otherwise the actual mesh calculation code here is much simpler as I know that each line is simply a single line segment with only two endpoints. I don't need to worry about any of the bevel join / miter join stuff like I have to do in the other PR if we have lines with multiple line segments

File: examples/add_vectors_image.py
Comment (sofroniewn) [reply examples/add_vectors_image.py]: Nope! I get the following error
```python
    image = 0.2*np.random.rand((24, 100)) + 0.5
  File "mtrand.pyx", line 1365, in mtrand.RandomState.rand
  File "mtrand.pyx", line 861, in mtrand.RandomState.random_sample
  File "mtrand.pyx", line 167, in mtrand.cont0_array
TypeError: 'tuple' object cannot be interpreted as an integer
```

Comment (sofroniewn) [reply examples/add_vectors_image.py]: ah - i see the difference now - fixed :-)

Comment (jni) [reply examples/add_vectors_image.py]: =)

File: napari/layers/_vectors_layer/model.py
Comment (sofroniewn) [reply napari/layers/_vectors_layer/model.py]: moved

**Comment by sofroniewn** [comment]
I have made the requested changes. Does @bryantChhun want to have a look before I merge? One thing that may have changed is that line widths are now in image coordinates not screen coordinates, so all your lines may look thicker than before, but it should be easy to fix

**Pull request merged**
